src/Redux/Reducers/authSlice.js

```diff
@@ -0,0 +1,63 @@
+import {createAsyncThunk, createSlice} from "@reduxjs/toolkit";
+import axios from "axios";
+import { AlertToast,SuccessToast } from "../../components/toasts";
+
+const initialState = {
+    currentUser:{},
+};
+
+export const login = createAsyncThunk("/auth/login", async (userDetail) =>{
+    try {
+        const response = await axios.post(`api/auth/login`,JSON.stringify(userDetail));
+        SuccessToast("Login Successful")
+        return response.data;
+        
+    } catch (error) {
+        console.log(error)
+        AlertToast(`${error}`)
+    }
+})
+
+export const signup = createAsyncThunk("/auth/signup", async (userDetail) => {
+    try{
+        const response = await axios.post("/api/auth/signup",JSON.stringify(userDetail))
+        SuccessToast("SignIn Successful")
+        return response.data
+    }
+    catch(error){
+        AlertToast(`${error}`)
+
+    }
```

Comment:
you can check https://redux-toolkit.js.org/api/createAsyncThunk#handling-thunk-errors for better error handling with thunk.

Reply:
Yep, actually will do in next pr